src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -229,13 +235,14 @@ public static MemoryMappedFile CreateFromFile(string path, FileMode mode, string
 
             if (capacity == DefaultSize)
             {
-                capacity = fileStream.Length;
+                capacity = fileSize;
```

Comment:
Thanks for pointing this out. 

To be honest I am not sure how to deal with it as a correct implementation of `Length` should return the same value before and after calling `Flush`, similar to our [implementation](https://github.com/dotnet/runtime/blob/a991e1349e4325770d8e2ddf085184a68cd2555b/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/BufferedFileStreamStrategy.cs#L55-L71)

```cs
long before = fileStream.Length;
fileStream.Flush();
long after = fileStream.Length;
Assert.Equal(before, after);
```

I can:

1. keep the code as it is (invalid `FileStream` implementation can in theory run into trouble)
2. move the length fetching and validation after call to `Flush` (it could change the exception thrown for empty files for cases where uses has also provided wrong `access` and `inheritability` which should be very rare).
3. re-fetch the `Length` after the call to `Flush` for custom types derived from `FileStream`

My preferences: 1 > 2 > 3

@stephentoub thoughts?

Reply:
We can go with your preference.  It's unlikely to actually be a problem.